### DISCUSSION TITLE
Is Factor framework 2.0 thread safe?

### DISCUSSION THREAD

**Discussion by Suresh-Vialto** [hmlongco/Factory#191] [General]
If resolve() and register { } methods are called at same time from different threads, is it thread safe? Will there any data inconsistencies issues?

extension Container {
    var names: Factory<[String]> {
        self { [] }
    }
}
struct Base {
    func getNames() -> [String] {
        return Container.shared.names.resolve()
    }
    func setNames(contacts: [String]) {
        Container.shared.names.register { contacts }
    }
}

**Comment by hmlongco** [comment]
It's thread safe. One request will lock and block the other until the first one completes.
